wallet/udb/common_test.go

```diff
@@ -45,31 +47,37 @@ func hexToBytes(origHex string) []byte {
 
 // createEmptyDB is a helper function for creating an empty wallet db.
 func createEmptyDB() error {
-	_ = os.Remove(emptyDb)
+	testDir, err := ioutil.TempDir("", testDir)
+	if err != nil {
+		return err
+	}
+
+	dbPath = filepath.Join(testDir, emptyDb)
+	db, err := walletdb.Create("bdb", dbPath)
+	defer db.Close()
 
-	db, err := walletdb.Create("bdb", emptyDb)
 	if err != nil {
-		return fmt.Errorf("unexpected error: %v", err)
+		return err
 	}
 
-	err = Initialize(db, &chaincfg.TestNet3Params, seed,
-		pubPassphrase, privPassphrase)
+	err = Initialize(db, &chaincfg.TestNet3Params, seed, pubPassphrase, 
+		privPassphrase)
 	if err != nil {
-		return fmt.Errorf("unexpected error: %v", err)
+		return err
 	}
 
 	err = Upgrade(db, pubPassphrase, &chaincfg.TestNet3Params)
 	if err != nil {
-		return fmt.Errorf("unexpected error: %v", err)
+		return err
 	}
 
-	return db.Close()
+	return nil
 }
 
 // cloneDB makes a copy of an empty wallet db. It returns a wallet db, store, a
 // stake store and a teardown function.
-func cloneDB(cloneName string) (walletdb.DB, *Manager, *Store, *StakeStore, func(), error) {
-	file, err := ioutil.ReadFile(filepath.Join("testdata", emptyDb))
+func cloneDB(sourcePath string, cloneName string) (walletdb.DB, *Manager, *Store, *StakeStore, func(), error) {
```

Comment:
We don't need to modify this function signature, instead it can just use the global that was already set up.